### PR TITLE
Fix gonefishing scenario

### PR DESCRIPTION
**Pull request by Xen0Xys**
Add 64 anvils to player
Add 100000 xp levels to player
Add max rod's enchants
Add version check for maximum compatibility
Fix work in 1.8.9 and 1.15.2, no tests on others versions

### CODE REVIEW

File: src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java [hunk 1]
```diff
@@ -4,27 +4,78 @@
...
+        rod.addUnsafeEnchantment(Enchantment.LURE, 8);
+        rod.addUnsafeEnchantment(Enchantment.LUCK, 255);
+
+        ItemMeta meta = rod.getItemMeta();
+        // Set item unbreakable code change in 1.15 and older versions
+        if(isMC114OrOlder()){
+            meta.spigot().setUnbreakable(true);
+        }else{
+            meta.setUnbreakable(true);
+        }
```
Comment (Mezy) [inline src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:61]: This part should be handled by VersionUtils.

File: src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java [hunk 2]
```diff
@@ -4,27 +4,78 @@
...
+
+        meta.setDisplayName("" + ChatColor.RED + "The Road");
```
Comment (Mezy) [inline src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:63]: Make messages translatable in the lang.yml please.

File: src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java [hunk 3]
```diff
@@ -4,27 +4,78 @@
...
+        rod.setItemMeta(meta);
 
         for (UhcPlayer uhcPlayer : e.getPlayersManager().getOnlinePlayingPlayers()){
             try {
                 uhcPlayer.getPlayer().getInventory().addItem(rod);
+                // Give player 10000 xl levels
+                uhcPlayer.getPlayer().setLevel(10000);
+                // Give player 64 anvils
+                ItemStack anvil = new ItemStack(Material.ANVIL);
+                anvil.setAmount(64);
```
Comment (Mezy) [inline src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:73]: Can be changed to 
```
ItemStack anvils = new ItemStack(Material.ANVIL, 64);
```
And I'd move it out of the for loop. Same as how the rod is created.

File: src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java [hunk 4]
```diff
@@ -4,27 +4,78 @@
...
 import com.gmail.val59000mc.scenarios.ScenarioListener;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
 import org.bukkit.Material;
 import org.bukkit.enchantments.Enchantment;
 import org.bukkit.event.EventHandler;
 import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import java.awt.*;
```
Comment (Mezy) [inline src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:15]: Is this import used?

**Comment by Xen0Xys** [comment]
Fixed needed changes

File: src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java [hunk 5]
```diff
@@ -1,30 +1,51 @@
...
+        rod.addUnsafeEnchantment(Enchantment.LUCK, 255);
+
+        ItemMeta meta = rod.getItemMeta();
+        // Set item unbreakable code change in 1.15 and older versions
+
+        if(UhcCore.getVersion() < 15){
+            meta.spigot().setUnbreakable(true);
+        }else{
+            meta.setUnbreakable(true);
+        }
```
Comment (Mezy) [inline src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:30]: Please move version specific code to VersionUtils.

File: src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java [hunk 6]
```diff
@@ -1,30 +1,51 @@
...
+
+        meta.setDisplayName("" + ChatColor.RED + "The Road"); // No need translation, it's a word play with an item name (rod / road)
```
Comment (Mezy) [inline src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:32]: Either remove the custom name or make it translatable. The fact that it's only one word doesn't mean it doesn't need to be translated.

File: src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java [hunk 7]
```diff
@@ -1,30 +1,51 @@
...
+        rod.setItemMeta(meta);
+
+        ItemStack anvil = new ItemStack(Material.ANVIL, 64);
```
Comment (Mezy) [inline src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:35]: Not too important but quick to change:
It's 64 so it's ``anvils`` not ``anvil``

File: src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java [hunk 8]
```diff
@@ -1,30 +1,51 @@
...
 
         for (UhcPlayer uhcPlayer : e.getPlayersManager().getOnlinePlayingPlayers()){
             try {
                 uhcPlayer.getPlayer().getInventory().addItem(rod);
+                // Give player 10000 xl levels
+                uhcPlayer.getPlayer().setLevel(10000);
+                // Give player 64 anvils
+
+                uhcPlayer.getPlayer().getInventory().addItem(anvil);
```
Comment (Mezy) [inline src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:44]: Not too important but quick to change:
This comment in for giving the anvils so don't put a empty line between them.
Maybe a empty line above the comment.

File: src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java (hunk 5 above)
Comment (Xen0Xys) [reply src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:30]: I am sorry but I don't know how I can do that, it is my first collaboration to open source project...

Comment (Mezy) [reply src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:27]: Alright, than dw. I'll change that part.

Comment (Xen0Xys) [reply src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/GoneFishingListener.java:27]: Will the code be added to the plugin?

**Pull request merged**